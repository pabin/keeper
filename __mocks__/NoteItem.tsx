import React, { ReactElement } from 'react';

const NoteItem = (): ReactElement => {
  return <></>;
};

export default NoteItem;
