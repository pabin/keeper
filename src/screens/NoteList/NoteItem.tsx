import React from 'react';
import { StyleSheet, TouchableOpacity, Text, View } from 'react-native';
import Icon from 'react-native-vector-icons/dist/FontAwesome';

import { colors } from '../../styles/colors';

const NoteItem = ({ note, navigation }) => {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={() => navigation.navigate('NoteDetail', { note })}
    >
      <View style={styles.iconContainer}>
        <Icon name="sticky-note" size={24} color={colors.white} />
      </View>
      <View style={styles.noteContainer}>
        <Text style={styles.text}>{note.title}</Text>
        <Text style={styles.text}>{note.createdAt}</Text>
      </View>
      <Icon name="chevron-right" size={20} color={colors.white} />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 16,
    margin: 12,
    elevation: 5,
    marginVertical: 8,
    borderRadius: 12,
    alignItems: 'center',
    flexDirection: 'row',
    backgroundColor: colors.lightBlue800,
  },

  iconContainer: {
    marginRight: 12,
  },

  noteContainer: {
    flex: 3,
  },

  text: {
    fontSize: 16,
    fontWeight: '600',
    color: colors.white,
  },
});

export default NoteItem;
