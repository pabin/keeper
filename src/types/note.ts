export type Note = {
  id: number;
  title: string;
  body: string;
  isFavourite: boolean;
  isArchived: boolean;
  createdAt: Date;
  updatedAt: Date;
};
