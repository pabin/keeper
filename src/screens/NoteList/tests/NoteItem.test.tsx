import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react-native';

import NoteItem from '../NoteItem';
import { note } from '../../../fixtures/note';

describe('NoteItem', () => {
  afterEach(cleanup);

  const navigateMock = jest.fn();
  const navigation = {
    navigate: navigateMock,
  };

  test('renders correctly', () => {
    const { toJSON } = render(<NoteItem note={note} navigation={navigation} />);
    expect(toJSON()).toMatchSnapshot();
  });

  test('navigates correctly', () => {
    const { getByTestId } = render(
      <NoteItem note={note} navigation={navigation} />,
    );

    const detailNavigator = getByTestId('detailNavigator');
    fireEvent.press(detailNavigator);
    expect(navigateMock).toHaveBeenCalledWith('NoteDetail', { note });

    const createNoteNavigator = getByTestId('createNoteNavigator');
    fireEvent.press(createNoteNavigator);
    expect(navigateMock).toHaveBeenCalledWith('CreateNotes', { note });
  });
});
