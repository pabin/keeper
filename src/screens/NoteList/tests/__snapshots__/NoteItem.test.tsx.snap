// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteItem renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 12,
      "flex": 1,
      "flexDirection": "row",
      "margin": 12,
      "marginVertical": 8,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 6,
        "width": 0,
      },
      "shadowOpacity": 0.2,
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#D6DBDF",
        "borderBottomLeftRadius": 12,
        "borderTopLeftRadius": 12,
        "flex": 4,
        "flexDirection": "row",
        "opacity": 1,
        "padding": 16,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 0.1,
      }
    }
    testID="detailNavigator"
  >
    <View
      style={
        Object {
          "marginRight": 12,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#607D8B",
              "fontSize": 24,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "#222",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        note title
      </Text>
      <Text
        style={
          Object {
            "color": "#222",
            "fontSize": 12,
          }
        }
      >
        Fri Jan 21 2022 | 11:34:45 AM
      </Text>
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "opacity": 1,
      }
    }
    testID="createNoteNavigator"
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#01579B",
            "fontSize": 22,
          },
          undefined,
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
